### PR TITLE
add tests for evm-7825 at boundaries

### PR DESCRIPTION
**Pull request by paologalligit**
# Description

Refer to the [ticket](https://github.com/vechain/protocol-board-repo/issues/1205) for the whole description of the EIP-7825 tests.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

File: consensus/consensus_integration_test.go [hunk 1]
```diff
@@ -108,3 +110,31 @@ func TestConsensus_StopsEnergyAtHardfork(t *testing.T) {
...
 }
+
+func TestVerifyBlock_EIP7825(t *testing.T) {
+	tests := []struct {
+		name         string
+		interstellar uint32
+		gas          []uint64
+	}{
+		{"tx over limit accepted before INTERSTELLAR", 2, []uint64{thor.MaxTxGasLimit + 1}},
+		{"txs at limit accepted after INTERSTELLAR", 1, []uint64{thor.MaxTxGasLimit, thor.MaxTxGasLimit}},
```
Comment (libotony) [inline consensus/consensus_integration_test.go]: Do we have tests around `txs over limit rejected after INTERSTELLAR`?

Comment (paologalligit) [reply consensus/consensus_integration_test.go]: yes, we have it in the e2e repo https://github.com/vechain/interstellar-e2e/blob/main/tests/eip7825/eip7825_test.go#L19-L26. We actually have also the accepted at exact `MaxTxGasLimit` there, I'll remove this one and leave only the pre forked

**Review by libotony — APPROVED** [review]

**Review by otherview — APPROVED** [review]
